### ISSUE TITLE
[Windows] Stale IPs

### ISSUE THREAD

**Issue by Pegasust** [canonical/multipass#2648]
**Describe the bug**
TL;DR: networking issue persists after purge if the VM is created under the same name.
I have an instance called `microk8s-master0` running just fine. For unknown reasons, after 1 day of uptime, `multipass list` yields this VM's ipv4 N/A. Recreating (after remove & purge) under any argument with the same name `microk8s-master0` will make the VM bad. Any other name works fine.

**To Reproduce**
I believe this is quite hard to reproduce reliably. I use Windows 10 with Hyper-V backend. Here are the steps:
1. `multipass launch --cpus "2" --disk "20G" --mem "3G" --name "microk8s-master0"  --cloud-init "C:\Users\_nhitrl\AppData\Local\Temp/microk8s.yaml" --network name="Guest OS to Host OS",mode="auto"`
2. Let it run for one day
3. `multipass info` now yields only microk8s-master0 to have N/A in the IPv4; if we did `multipass info --format json`, the IPv4 will be an empty list (`[]`)
4. `multipass remove microk8s-master0`
5. `multipass purge`
6. `multipass launch --cpus "2" --disk "20G" --mem "3G" --name "microk8s-master0"  --cloud-init "C:\Users\_nhitrl\AppData\Local\Temp/microk8s.yaml" --network name="Guest OS to Host OS",mode="auto"` -> Time out waiting for  instance to reply; return code != 0
7. `multipass info` yields microk8s-master0 IPv4 being N/A

Side note: I configured Guest OS to Host OS as a virtual switch that connects with the underlying ethernet hardware. Even if I did launch without any networking or cloud-init,

**Expected behavior**
Even if there was any error, purging and recreating the VM with the same name should put the VM in a clean state.

**Logs**
Please provide logs from the daemon, see [accessing logs](https://multipass.run/docs/accessing-logs) on where to find them on your platform.

**Additional info**
 - OS: Windows 10
- PS C:\Users\_nhitrl> multipass version
multipass   1.9.1+win
multipassd  1.9.1+win
- PS C:\Users\_nhitrl> multipass info --all
info failed: ssh connection failed: 'Timeout connecting to microk8s-master0.mshome.net'
PS C:\Users\_nhitrl> multipass info
PS C:\Users\_nhitrl> multipass list
Name                    State             IPv4             Image
primary                 Stopped           --               Ubuntu 20.04 LTS
kube-grandmaster        Stopped           --               Ubuntu 20.04 LTS
kube-master-1           Stopped           --               Ubuntu 20.04 LTS
kube-master-2           Stopped           --               Ubuntu 20.04 LTS
kube-worker-0           Stopped           --               Ubuntu 20.04 LTS
kube-worker-1           Stopped           --               Ubuntu 20.04 LTS
microk8s-master0        Running           N/A              Ubuntu 20.04 LTS
microk8s-master1        Running           172.20.3.203     Ubuntu 20.04 LTS
                                          192.168.0.74
                                          10.1.119.128
microk8s-worker0        Running           172.20.10.130    Ubuntu 20.04 LTS
                                          10.1.170.128

**Additional context**
- I tried redoing the creation many times, the error persists
- Apart from microk8s-master0, I also have microk8s-master1 (identically configured) and microk8s-worker0 (differently configured). Both of which ran fine with no network error (yet)
- Even if I launched microk8s-master0 without cloud-init or network, the error persists.
- This is not the first time I encounter this bug
[multipass.zip](https://github.com/canonical/multipass/files/9032619/multipass.zip)

**Comment by luis4a0** [comment]
On the other hand, can you please try letting Multipass create the bridge over the network hardware? That is, using `--network=Ethernet` or so?

**Comment by ricab** [comment]
Hi @Pegasust, are you perhaps being hit by the [stale IP issue](https://multipass.run/docs/troubleshoot-networking#heading--stale-internet-connection-sharing-lease) in `hosts.ics`.

**Comment by Pegasust** [comment]
Thank you for the insights, this really describes what the problem might be. I'll investigate as soon as I can about this.

**Comment by ricab** [comment]
Related to #2057

**Comment by ricab** [comment]
Duplicate of  #3354
